### PR TITLE
Add --allowed-address-type option

### PR DESCRIPTION
**Pull request by hiroya-uga**
Hello there,

(I used translators)

I am considering hosting the Nu Validator internally to validate the correctness of HTML.   
In order to mitigate the risk of communication outside our organization, I have introduced the `--allowed-address-type` option.

- `all` - Allows inspection of all URLs.
- `same-origin` - Allows inspection of URLs from the same origin.
- `none` - Prohibits HTML inspection from URLs.

I would be grateful if you would accept this change.

Best regards.

## As-Is

There are no restrictions on URLs.

## To-Be

It becomes possible to impose restrictions on URLs.

### `all` (Default) - No changed.

image

### `same-origin` - Allows inspection of URLs from the same origin.

Page load: 

\* No changed.

image

After request cross-origin URL:

image

### `none` - Prohibits HTML inspection from URLs.

Page load: 

image

Request via URL: 

\* address is disabled.

image

## How to test

https://github.com/hiroya-uga/validator#build-instructions

Run nu validator after build.

NOTE: If your OS is macOS, you have to use `python3` intead of `py` or `python`

```shell
python ./checker.py run 
```

```shell
python ./checker.py --allowed-address-type=same-origin run
```

```
python ./checker.py --allowed-address-type=none run
```

And then access to <http://0.0.0.0:8888/>

### CODE REVIEW

File: build/build.py [hunk 1]
```diff
@@ -1712,6 +1714,14 @@ def printHelp():
...
     print("  --bind-address=0.0.0.0     -- Sets server bind address")
     print("  --port=8888                -- Sets server port number")
+    print("  --allowed-address-type=<value>")
+    print("                                Set the value to")
+    print("                                specify URL acceptance.")
+    print("                                Possible values:")
+    print("                                - 'all': Accept all URLs (default)")
+    print("                                - 'same-origin': Accept only")
+    print("                                  same-origin URLs")
+    print("                                - 'any': Disallow address usage)")
```
Comment (sideshowbarker) [inline build/build.py:1724]:
```suggestion
    print("                                Set the value to specify")
    print("                                what URLs are accepted.")
    print("                                Possible values:")
    print("                                - 'all': Accept all URLs (default)")
    print("                                - 'same-origin': Accept only")
    print("                                  same-origin URLs")
    print("                                - 'any': Disallow URL input)")
```

File: build/build.py [hunk 2]
```diff
@@ -1712,6 +1714,14 @@ def printHelp():
...
     print("  --bind-address=0.0.0.0     -- Sets server bind address")
     print("  --port=8888                -- Sets server port number")
+    print("  --allowed-address-type=<value>")
+    print("                                Set the value to specify")
+    print("                                what URLs are accepted.")
+    print("                                Possible values:")
+    print("                                - 'all': Accept all URLs (default)")
+    print("                                - 'same-origin': Accept only")
+    print("                                  same-origin URLs")
+    print("                                - 'any': Disallow URL input)")
```
Comment (sideshowbarker) [inline build/build.py:1724]: Actually, shouldn’t this value be `'none'`?

File: build/build.py [hunk 3]
```diff
@@ -1712,6 +1714,14 @@ def printHelp():
     print("  --name=Validator.nu        -- Sets service name")
     print("  --bind-address=0.0.0.0     -- Sets server bind address")
     print("  --port=8888                -- Sets server port number")
+    print("  --allowed-address-type=<value>")
+    print("                                Set the value to specify")
+    print("                                what URLs are accepted.")
```
Comment (sideshowbarker) [inline build/build.py:1719]:
```suggestion
    print("  --allowed-address-type=<value>")
    print("                                Sets which URLs the checker allows.")
```

File: build/build.py [hunk 4]
```diff
@@ -1712,6 +1714,13 @@ def printHelp():
     print("  --name=Validator.nu        -- Sets service name")
     print("  --bind-address=0.0.0.0     -- Sets server bind address")
     print("  --port=8888                -- Sets server port number")
+    print("  --allowed-address-type=<value>")
+    print("                                Sets which URLs the checker allows.")
+    print("                                Possible values:")
+    print("                                - 'all': Accept all URLs (default)")
+    print("                                - 'same-origin': Accept only")
+    print("                                  same-origin URLs")
+    print("                                - 'any': Disallow URL input)")
```
Comment (sideshowbarker) [inline build/build.py:1723]:
```suggestion
    print("                                - 'all': Allow all URLs (default)")
    print("                                - 'same-origin': Allow only")
    print("                                  same-origin URLs")
    print("                                - 'any': Disallow all URLs")
```

File: build/build.py (hunk 2 above)
Comment (hiroya-uga) [reply build/build.py]: @sideshowbarker Thank you for reviewing and telling me correct English.
Yes, I changed just now!

[cc50de6](https://github.com/validator/validator/pull/1638/commits/cc50de6f22cb90746bd892e62815bb4593f6dad8)

Comment (hiroya-uga) [reply build/build.py]: @sideshowbarker Sorry, It appears that unintended differences remain.
Please wait for fixing...

Comment (hiroya-uga) [reply build/build.py]: @sideshowbarker 
Thank you for your waitting!
All corrections are done. Please check it!

[5e4d10f](https://github.com/validator/validator/pull/1638/commits/5e4d10f710f8e73e62b4f6b1f8f2fb75667b998c)

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
@hiroya-uga Beautiful — thanks much

**Pull request merged**